### PR TITLE
Refactor air water vv

### PR DESCRIPTION
**Pull request by adimako**
@cekees @tridelat @Giovanni-Cozzuto-1989 @nerisaurus:
 The structure of air-water-vv is reorganised and the validation cases should be arranged in theme folders. I have removed some cases from the master branch but these eventually will have to go in, using context options, spatial tools and boundary conditions. In that sense, it might be better for example to use the new cases to derive them than start modifying the old cases. Eventually we will do the same for the 3D cases

### CODE REVIEW

**Comment by nerisaurus** [comment]
I like the restructuring.  Aside from the removal of some tests and the addition of others, will anything change #105 ?  That is, are any of the tests that were previously in master changed (aside from being moved - changed in ways that could break or fix them)?  If there's no changes, I can focus on checking out the new tests alone, and expect the old issues (or non-issues) to stay the same.

What makes a test a "benchmark"? Is this just a miscellaneous category, or do we have more data / more use for these tests as standard benchmarks for running a code like proteus?

**Comment by adimako** [comment]
@nerisaurus I have updated the issue with some notes. I think we should scrap the old set-up files but at the same time maintain the same cases along with descriptions etc. You can check the numerical tanks. It will also help if you populate a list with the old cases that need to be updated to the new standards.

I guess benchmark is a relatively short and simple test case, which simulates a fundamental process e.g. dambreak, sloshing, cavity flow (.... backward step, Sod's tube and many many others). In that sense the wave tank tests are also benchmark, but I think it is worth dedicating a separate test set for these.

**Comment by nerisaurus** [comment]
@adimako Thanks for the notes.   I'm slowly slogging through the Gauge-BC-Wave-Shape interconnections at the moment, but I hope to have my weir_tank shape extension finished this week.  If compatibility with the old solutions is not that big of a concern [0], that should take care of all the hydraulic structures except for the sluice gate [1].

For these, at least, I could actually merge them all into one file, and have the type of test as a context option (they all have a concept of "width" and "height" for a single obstacle, and should interact similarly with sponge layers, waves, and the other context options they have.  Is that desirable, or going a bit too far?

I've been using text files to hold context options for all the old tests (so something like 'parun ... -o "default_broad_crested_weir_AV.txt" ' [2] could set up a test that would follow the old `./broad_crested_weir_AV/` case).  I believe this works with `-v' as well, so we could set up defaults for each type of hydraulic structure while still allowing variation with `-v` (or by making a new text file).  I'll also probably extend this to the coarse/medium/fine splits in certain problems to begin with - but we might then slowly thin out less useful text options [3].

[0] By which I mean I've been following a sort of default policy of making sure the tests match the output of the previous tests exactly for each change up until a change which justifies breaking it.  Usually these changes are with the mess of boundary conditions which appear in older codes, and which don't have exact matches in BoundaryConditions and WaveTools.  This process is a bit slow

[1] Unless the top of the sluice can have a small opening.  The code just currently can't attach obstacles to the top of the tank.  If I replicate it by a box that almost touches the top, or write in a small addition to the code to let it pick up a top obstacle too, this also gets covered.

[2] It's probably valuable to stick to a default naming scheme here, in order to keep all the text files in one place.  In fact, it might even be a good idea to use a subdirectory for each case, if we want to continue this practice for all the problems (and then we can be more free with how we name each file).  Too many text files can clutter up the folder and make it hard to parse and navigate (especially since there's other text files everywhere).  I also believe that the file doesn't necessarily have to be `.txt`, as long as it reads correctly to parun.  I'm open to suggestions for the most conveniently readable file type to use.

[3] The coarse/medium/fine meshes are generally very small changes (sometimes only to one context option) which are easy to just change on the fly with `-v`.  They might be useful in some problems (because they make it so we don't have to think about what's a fine or coarse mesh in this context), but some probably aren't needed.

**Comment by nerisaurus** [comment]
Re @adimako comments:

Yeah, I meant `-C`.  I'm too used to typing `-v` everywhere now.

By a default naming scheme, I mean that the option files within a test directory could either stick to one common prefix or all be put in one common subdirectory (which is probably the better idea - although having common prefix patterns [so that all "coarse" grids stick in the same spot, for instance] could still be useful there), so that the files are easily found.  

I'm not sure if a .csv or some other kind of file might be handy for easily rewriting the context options, while still appearing to proteus in the right format to read the options from.

**Comment by cekees** [comment]
The context  options have to be a white-space  separated string with each string being a valid  string on the left hand side and a valid Python literal on the right hand side. So something like "g=[0,-9.81,0] a=1" gets split into two expressions, then the dictionary key "g" gets set  to the valid Python literal [0,-9.8,0], which is a standard Python array and "a" gets the value 1.  There wouldn't be much to gain by trying to support numpy arrays, since we're not trying to load  large arrays  through options.  I'm not saying we shouldn't rewrite how the context options are specified, I'm just explaining how it works. String -> split on white space -> list of strings with "=" -> split on "=" -> use left side as option name and process right hand side as Python literal using ast.literal_eval.  I would say as in incremental improvement we could just make the string process  more forgiving and more informative on failure. For example we could allow expressions that terminate with ";" and try to support more white space, maybe by splitting on "=" first and then processing the lhs and rhs.
